src/modules/announcement/announcement.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Inject, Injectable, NotFoundException } from '@nestjs/common';
+import { ANNOUNCEMENT_REPOSITORY } from 'src/core/constants';
+import { Announcement } from 'src/entities/announcement.entity';
+import { AnnouncementDTO } from './dto/announcement.dto';
+
+@Injectable()
+export class AnnouncementService {
+  constructor(
+    @Inject(ANNOUNCEMENT_REPOSITORY)
+    private announcementRepository: typeof Announcement,
+  ) {}
+
+  async findOneById(announcementId: number) {
+    try {
+      return await this.announcementRepository.findOne({
+        where: { id: announcementId },
+        rejectOnEmpty: true,
+      });
+    } catch (e) {
+      console.error(e);
+      throw new NotFoundException();
+    }
+  }
+
+  async findAll() {
+    return this.announcementRepository.findAll({ order: ['id'] });
+  }
+
+  async findShown() {
+    return this.announcementRepository.findAll({
+      where: { show: true },
+      order: ['id'],
```

Comment:
I think it might be great if set order type like `order: [['id','ASC']]` or `order: [['id','DESC']]`. because it easy to read than set only `order: ['id']`